python/lsst/pipe/tasks/selectImages.py

```
@@ -339,3 +339,125 @@ def runDataRef(self, dataRef, coordList, makeDataRefList=True, selectDataList=[]
+            Result struct with components:
+            - ``exposureList``: the selected exposures
+                (`list` of `lsst.pipe.tasks.selectImages.BaseExposureInfo`).
+            - ``dataRefList``: the optional data references corresponding to
+                each element of ``exposureList``
+                (`list` of `lsst.daf.persistence.ButlerDataRef`, or `None`).
+
+        Notes
+        -----
+        We use the "convexHull" method of lsst.sphgeom.ConvexPolygon to define
+        polygons on the celestial sphere, and test the polygon of the
+        patch for overlap with the polygon of the image.
+
+        We use "convexHull" instead of generating a ConvexPolygon
+        directly because the standard for the inputs to ConvexPolygon
+        are pretty high and we don't want to be responsible for reaching them.
+        """
+        if self.config.nImagesMax <= 0:
+            raise RuntimeError(f"nImagesMax must be greater than zero: {self.config.nImagesMax}")
+
+        psfSizes = []
+        dataRefList = []
+        exposureInfoList = []
+
+        patchVertices = [coord.getVector() for coord in coordList]
+        patchPoly = lsst.sphgeom.ConvexPolygon.convexHull(patchVertices)
+
+        for data in selectDataList:
+            dataRef = data.dataRef
+            imageWcs = data.wcs
+            cal = dataRef.get("calexp", immediate=True)
+            psfSize = cal.getPsf().computeShape().getDeterminantRadius()
+            nx, ny = cal.getDimensions()
+
+            imageBox = afwGeom.Box2D(afwGeom.Point2D(0, 0), afwGeom.Extent2D(nx, ny))
+            try:
+                imageCorners = [imageWcs.pixelToSky(pix) for pix in imageBox.getCorners()]
+            except (pexExceptions.DomainError, pexExceptions.RuntimeError) as e:
+                # Protecting ourselves from awful Wcs solutions in input images
+                self.log.debug("WCS error in testing calexp %s (%s): deselecting", dataRef.dataId, e)
```

This amounts to silently dropping problematic WCS (and probably other things, given how broad the catch clause is). If you're not willing to abort, at least print a warning or higher.

---

For both of these we want to be permissive--the goal is to get a usable coadd, not to guarantee that all images can be processed.

---

It's not ignoring images itself that bothers me, it's the fact that it happens almost silently. Unless you mean that dropping images in one of these two cases is a normal occurrence, not just an acceptable one?

---

With the refactor this is now upstream in `WcsSelectImagesTask`...